pulsar-functions/java-examples/src/main/java/org/apache/pulsar/functions/api/examples/PublishFunctionWithMessageConf.java

```diff
@@ -40,13 +41,14 @@ public Void process(String input, Context context) {
         properties.put("input_topic", context.getCurrentRecord().getTopicName().get());
         properties.putAll(context.getCurrentRecord().getProperties());
 
-        Map<String, Object> messageConf = new HashMap<>();
-        messageConf.put(TypedMessageBuilder.CONF_PROPERTIES, properties);
-        if (context.getCurrentRecord().getKey().isPresent()) {
-            messageConf.put(TypedMessageBuilder.CONF_KEY, context.getCurrentRecord().getKey().get());
+        try {
+            TypedMessageBuilder<String> msgBuilder = context.newOutputMessage(publishTopic, null);
+            msgBuilder.value(output).properties(properties).
+                    key(context.getCurrentRecord().getKey().get()).
+                    eventTime(System.currentTimeMillis()).sendAsync();
+        } catch (PulsarClientException e) {
+            e.printStackTrace();
```

Comment:
can we use a logger to log this

Reply:
@wolfstudy can you replace this with context.getLogger().error(e)

Reply:
thanks @jerrypeng remind